stl/inc/optional

```diff
@@ -612,6 +612,11 @@ template <class _Ty>
 optional(_Ty) -> optional<_Ty>;
 
 template <class _Ty1, class _Ty2>
+#ifdef __cpp_lib_concepts
+    requires requires(const _Ty1& _Elem1, const _Ty2& _Elem2) {
+        { _Elem1 == _Elem2 } -> _Implicitly_convertible_to<bool>;
+    }
+#endif // __cpp_lib_concepts

```

Comment:
Consider making these trailing requires clauses so you can use the function parameters, e.g. this would become
```c++
template <class _Ty1, class _Ty2>
_NODISCARD constexpr bool operator==(const optional<_Ty1>& _Left, const optional<_Ty2>& _Right) noexcept(
    noexcept(_Fake_copy_init<bool>(*_Left == *_Right))) // strengthened
#ifdef __cpp_lib_concepts
    requires requires { { *_Left == *_Right } -> _Implicitly_convertible_to<bool>; } 
#endif  // __cpp_lib_concepts
    {
```